### PR TITLE
[ci skip] Add rake task to cleanup merged (remote) branches

### PR DESCRIPTION
**Pull request by basiszwo**

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://basiszwo-git-cleanup-task.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://basiszwo-git-cleanup-task.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
For upstream I created a second refspec that pulls all GitHub-PRs (this is a nice convenience feature offered by github):

```
fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*
```

However, now this job will try to prune all of them (I was not keen enough to see what happens if I press "yes"):

```
upstream/pr/1
upstream/pr/10
upstream/pr/100
upstream/pr/1000
upstream/pr/1001
upstream/pr/1002
```

Generally, this job has a quite large scope, e.g. I also added remotes for various other contributors, to be able to pull their branches directly. I don't want to delete other peoples remote branches (even if I was added as collaborator for their forks):

```
floriank/hotfix/fix_wp_csv_export
floriank/hotfix/syntax_highlight
machisuji/feature/api_v3_basic_auth_api_doc
machisuji/feature/basic_auth
```

IMO this job should be limited to be deleting from the remote called `origin`, since this is by convention the name of your own fork of a repository.

File: lib/tasks/git.rake [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+namespace :git do
+  desc 'Clean up your locale and remote repository'
+  task :clean do
+    # FIXME: change this to master once we are there
```
Comment (NobodysNightmare) [inline lib/tasks/git.rake:33]: I see you have visions ;-)

**Comment by NobodysNightmare** [comment]
Btw. :+1: for the general idea of this task!

**Comment by basiszwo** [comment]
@NobodysNightmare applied your feedback. can you please check for your repo and merge if good?

File: lib/tasks/git.rake [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                        !b.starts_with?('origin') ||
+                        excluded_branches.include?(b.split('/').drop(1).join('/'))
+                      }
+
+    local_branches = `git branch --merged`
+                     .gsub(/^\* /, '')
+                     .split("\n")
+                     .map(&:strip)
+                     .reject{ |b|
+                       !b.starts_with?('origin') ||
```
Comment (NobodysNightmare) [inline lib/tasks/git.rake]: For me `git branch --merged` will never output the prefix (`origin`). I think you should only apply this condition for remote branches (`git branch -r --merged`).

Sample outputs from my console:

Local branches:

```
$ git branch --merged
  feature/aggregated_diffs
  feature/aggregated_mails
```

remote branches:

```
$ git branch -r --merged
  origin/dev
  origin/feature/19490-select2-3.5-upgrade
  origin/feature/aggregated_diffs
  origin/feature/aggregated_mails
```

edit: Yup, just verified that **the rake task will not remove any local branches** now...

**Comment by basiszwo** [comment]
Rebased with current rails 4 dev branch and fixed the local branch issue.

**Pull request merged**